### PR TITLE
[GAL-359] Tezos Error Handler

### PR DESCRIPTION
**Pull request by jakzaizzat**
Added error handler for tezos authentication flow.

**Scenario**
- [x] User closed the modal
- [x] Different network 
- [x] Reject signature

**Demo**

**User closed the beacon modal** 
https://user-images.githubusercontent.com/4480258/189296970-56520a5c-607d-49c8-8126-12f4002d777f.mp4

**Reject the signature**

https://user-images.githubusercontent.com/4480258/189307992-313d33c8-9e7d-46ed-87a0-a092ab587680.mp4

### CODE REVIEW

File: src/contexts/beacon/BeaconContext.tsx [hunk 1]
```diff
@@ -36,12 +42,33 @@ type Props = { children: ReactNode };
 let beaconClient: DAppClient;
 
 if (typeof window !== 'undefined') {
-  beaconClient = new DAppClient({ name: 'Gallery' });
+  beaconClient = new DAppClient({
+    name: 'Gallery',
+    preferredNetwork: NetworkType.MAINNET,
```
Comment (jakzaizzat) [inline src/contexts/beacon/BeaconContext.tsx:47]: Set the `mainnet` as default so there is no mismatch network

File: src/components/WalletSelector/multichain/tezos/tezosUtils.ts [hunk 2]
```diff
@@ -19,3 +20,9 @@ export function getNonceNumber(nonce: string) {
   const splittedNonceMessage = nonce.split(' ');
   return splittedNonceMessage[splittedNonceMessage.length - 1];
 }
+
+export function getErrorCode(error: BeaconError) {
+  // https://typedocs.walletbeacon.io/enums/beaconerrortype.html
+  // Example: [ABORTED_ERROR]:The action was aborted by the user.
+  return error.message.split(':')[0].replace('[', '').replace(']', '');
```
Comment (jakzaizzat) [inline src/components/WalletSelector/multichain/tezos/tezosUtils.ts:27]: The beacon error didnt provide any `code` properties in their response just like ethereum. 

Need to trim their error message to get their unique code 🤦🏻‍♂️

Comment (Robinnnnn) [reply src/components/WalletSelector/multichain/tezos/tezosUtils.ts:27]: hehehe

**Review by Robinnnnn — APPROVED** [review]
thank you, jakz! this is incredible!!!

for some reason e2e failed because the metamask pop-up no longer appears according to this test run 🤔 https://dashboard.cypress.io/projects/igfgns/runs/51743924-ea78-4511-9ddd-2f8ca3246625/test-results/537bf29e-31df-4a05-b10f-9a3aeaf160f6/video

i pulled the branch down locally and couldn't repro though, so i'm wondering if it's a flake

either way, the PR LGTM

**Comment by jakzaizzat** [comment]
Sweet @Robinnnnn !

Seems like the third e2e test case always flaky because of the metamask pop-up 🤦🏻‍♂️.  

Its quite known issue on synpress and happens on several users (expecially on ci)

**Pull request merged**